/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto;

message Condition {
  enum ConditionType {
    // Use alias to make enum name consistent with others
    option allow_alias = true;
    succeeded = 0;
    SUCCEEDED = 0;
  }
  enum ConditionStatus {
    // Use alias to make enum name consistent with others
    option allow_alias = true;
    Unknown = 0;
    UNKNOWN = 0;
    True = 1;
    TRUE = 1;
    False = 2;
    FALSE = 2;
  }
  ConditionType type = 1;
  ConditionStatus status = 2;
  string last_transition_time = 3;
  string reason = 4;
  string message = 5;
}
